packages/lodestar/src/api/impl/beacon/state/state.ts

```diff
@@ -44,17 +51,42 @@ export class BeaconStateApi implements IBeaconStateApi {
     };
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   async getStateValidators(stateId: StateId, filters?: IValidatorFilters): Promise<phase0.ValidatorResponse[]> {
     const state = await resolveStateId(this.chain, this.db, stateId);
     if (!state) {
       throw new StateNotFound();
     }
-    //TODO: implement filters
+
+    const currentEpoch = getCurrentEpoch(this.config, state);
+
+    const validators: phase0.ValidatorResponse[] = [];
+    if (filters?.indices) {
+      for (const id of filters.indices) {
+        const validatorIndex = getStateValidatorIndex(id, state, this.chain);
+        if (validatorIndex) {
+          const validator = state.validators[validatorIndex];
+          if (filters.statuses && !filters.statuses.includes(getValidatorStatus(validator, currentEpoch))) {
+            continue;
+          }
+          const validatorResponse = toValidatorResponse(
+            validatorIndex,
+            validator,
+            state.balances[validatorIndex],
+            currentEpoch
+          );
+          validators.push(validatorResponse);
+        }
+      }
+      return validators;
+    } else if (filters?.statuses) {
```

Comment:
If there is indices filter, I'm pretty sure we won't go trough this code path and filter by status as well.

You should add test case

Reply:
I assumed your comment here (https://github.com/ChainSafe/lodestar/pull/2234#discussion_r605451789) was implying that we needed to account for when both filter types are passed in.  I've tested it locally, but I'll add a formal test case as well.

Reply:
Either way, the current structure does allow for both filter types to be passed in.

Reply:
Also, this particular line you pointed out could also account for if only the statuses are passed in.  Still, I'm adding more tests now.  Also just realized there were two tests that had a `skip` on them for `getStateValidators`, so I will enable those as well and modify as needed.

Reply:
Maybe I'm mistaken but if you pass both indices and status filter. Isn't this code gonna return all validators for given indices but it won't filter them by status?

Reply:
ah, nvm, missed some code